### PR TITLE
15 comments update and destroy apis with tests

### PR DESCRIPTION
**Pull request by W-Shih**
- feature: add update and destroy comment apis
- test: add tests for update and destroy comment apis
- feature: add id to admin page

### CODE REVIEW

File: comments/api/tests.py [hunk 1]
```diff
@@ -110,3 +114,101 @@ def test_create_api(self):
...
             'tweet does not exist.'
         )
+
+    def test_update_api(self):
+        comment = self.create_comment(self.lbj23, self.tweet, 'GOAT!')
+        url = COMMENT_DETAIL_URL.format(comment.id)
+
+        # Non-login users
+        response = self.anonymous_client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
```
Comment (wen-wayne-shih) [inline comments/api/tests.py:124]: This permission_class might be not suitable when retrieve api is considered.
Can you add some comment on this?

File: comments/api/tests.py [hunk 2]
```diff
@@ -110,3 +114,101 @@ def test_create_api(self):
...
+
+        response = self.anonymous_client.put(url, {'content': 'dummy comment'})
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+        # Not PUT method
+        response = self.kd35_client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_405_METHOD_NOT_ALLOWED)
```
Comment (wen-wayne-shih) [inline comments/api/tests.py:131]: This permission_class might be not suitable when retrieve api is considered.
Can you add some comment on this?

**Review by wen-wayne-shih — CHANGES_REQUESTED** [review]
This permission_class might be not suitable when retrieve api is considered.
Can you add some comment on this?

File: comments/api/tests.py (hunk 1 above)
Comment (W-Shih) [reply comments/api/tests.py]: Resolved as per suggestion!

File: comments/api/tests.py (hunk 2 above)
Comment (W-Shih) [reply comments/api/tests.py]: Resolved as per suggestion!

File: comments/api/tests.py [hunk 3]
```diff
@@ -110,3 +114,101 @@ def test_create_api(self):
...
+        self.assertNotEqual(comment.created_at, now)
+        self.assertNotEqual(comment.updated_at, old_updated_at)
+
+    def test_delete_api(self):
+        comment = self.create_comment(self.lbj23, self.tweet, 'GOAT!')
+        url = COMMENT_DETAIL_URL.format(comment.id)
+
+        # Non-login users
+        response = self.anonymous_client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
```
[the remark this answers is not in the document]
Comment (W-Shih) [reply comments/api/tests.py:204]: Resolved as per suggestion!

File: comments/api/tests.py [hunk 4]
```diff
@@ -110,3 +114,101 @@ def test_create_api(self):
...
+
+        response = self.anonymous_client.delete(url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+        # Not DELETE method
+        response = self.kd35_client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_405_METHOD_NOT_ALLOWED)
```
[the remark this answers is not in the document]
Comment (W-Shih) [reply comments/api/tests.py:214]: Resolved as per suggestion!

File: comments/api/views.py [hunk 5]
```diff
@@ -19,12 +20,19 @@
...
+    CommentSerializerForUpdate,
+)
+from comments.models import Comment
 
 
 class CommentViewSet(viewsets.GenericViewSet):
+    queryset = Comment.objects.all()
     serializer_class = CommentSerializerForCreate
-    permission_classes = [IsAuthenticated]
+    permission_classes = [IsAuthenticated, IsObjectOwner]
```
[the remark this answers is not in the document]
Comment (W-Shih) [reply comments/api/views.py:38]: Resolved as per suggestion!

**Review by W-Shih — COMMENTED** [review]
Resolved as suggestions!

**Review by wen-wayne-shih — APPROVED** [review]

**Pull request merged**